### ISSUE TITLE
Neovim issues

### ISSUE THREAD

**Issue by alex-ameen** [l04m33/vlime#39]
Hey I’m loving the plugin, and it runs pretty smoothly in Neovim. 
There are some small issues though. For instance the interactive mode is non-functional. I can’t precisely track down the cause but I have a feeling its an issue with remapping <CR>, if this is the case i think it should be a relatively easy fix. 

Clearing the REPL often dumps errors, i think it is trying to evaluate parts of the buffer as vimcript?

Finally REPL and other additional windows do not “scroll to bottom” when new linea appear, i usually need to swap my window and jump to the bottom of the buffer with G.

On the whole i absolutely love the plugin and want to thank you for your incredible work. 
Im more than happy to put in time to put in time to fix these for a merge; I just wanted to check in for some direction.

**Comment by Fedreg** [comment]
@aakropotkin Just started using this plugin on neovim and noticed the same issues.  Were you ever able to find a solution?  thx!

**Comment by alex-ameen** [comment]
Whoa blast from the past.

I switched to Emacs because of this issue which probably isn't helpful news to report back. After years of arm wrestling with Vimscript I finally got fed up with it 😂.

I don't remember if i managed to fix this though, I can look at my config and see if anything jumps out at me.

If you could refresh my memory with some of the errors that may help me remember.

**Comment by alex-ameen** [comment]
Found this in my config

```
" Vlime Buffer Fix: ----------------------------------------------------- {{{2
 
augroup CustomVlimeInputBuffer
  autocmd!
  autocmd FileType vlime_input inoremap <silent> <buffer> <Tab> <C-R>=vlime#plugin#VlimeKey( "tab" )<CR>
  autocmd FileType vlime_input setlocal omnifunc=vlime#plugin#CompleteFunc
  autocmd FileType vlime_input setlocal indentexpr=vlime#plugin#CalcCurIndent()
augroup end
 
" End Vlime Buffer Fix -------------------------------------------------- }}}2
```

I also have a `vlime` folder under `~/.config/nvim`, I don't remember if that's normal, so i might have forked and modified it there.

If that snippet doesn't work i can upload my "fork".

**Comment by Fedreg** [comment]
@aakropotkin Thank you!!

I also use Emacs and slime so not the end of the world if I can't get this to play nice with neovim.

I'll try your snippet and see if that helps.  I don't mind the vimscript so much if I need to dig into that a bit..   Just saw your issue earlier and figured it was worth a shot to ask you. 

Appreciate the quick response!!

**Comment by phmarek** [comment]
Well, do you mean "interactive mode" as in "typing directly in the REPL buffer"? There's a PR for that: https://github.com/vlime/vlime/pull/55

**Comment by alex-ameen** [comment]
Yeah I believe that is what I was talking about. Again this was a long time ago so my memory is hazy.

**Comment by equwal** [comment]
This issue is still relevant.

I can't get the input buffer to print anything other than `--` and compiler errors (so I have a `rel` not a `repl`.)

Good news: the code snippet above is magic and it works now.

**Comment by equwal** [comment]
Also applies to vim though.
